tenpy/algorithms/dmrg.py

```diff
@@ -185,215 +207,52 @@ class     A string stands for one of the engines defined in this module,
     }
 
 
-class Engine(NpcLinearOperator):
-    """Prototype for an DMRG 'Engine'.
+class DMRGEngine(Sweep):
+    """ Generic 'Engine' for the single-site DMRG algorithm.
 
-    This class is the working horse of DMRG. It implements the :meth:`sweep` and large
-    parts of the (two-site) optimization.
-    During the diagonalization (i.e. after calling :meth:`prepare_diag`), the class represents
-    the effective two-site Hamiltonian, which looks like this::
-
-        |        .---            ----.
-        |        |     |      |      |
-        |        LP----W[i0]--W[i1]--RP
-        |        |     |      |      |
-        |        .---            ----.
-
-    `LP` and `RP` are left and right parts of the :class:`~tenpy.networks.mpo.MPOEnvironment`,
-    `W[i0]` and `W[i1]` are the MPO matrices of the Hamiltonian at the two sites ``i0, i1=i0+1``.
-    How this network is then actually contracted in detail is left to derived classes.
+    This engine is implemented as a subclass of 
+    :class:`~tenpy.algorithms.mps_sweeps.Sweep`. It contains all methods that
+    are generic between :class:`OneSiteDMRGEngine` and :class:`TwoSiteDMRGEngine`.
 
     Parameters
     ----------
-    psi : :class:`~tenpy.networks.mps.MPS`
-        Initial guess for the ground state, which is to be optimized in-place.
+    EffectiveH : class type
+        Class for the effective Hamiltonian (i.e., a subclass of
+        :class:`~tenpy.algorithms.mps_sweeps.EffectiveH`. Has a `length` class attribute which
+        specifies the number of sites updated at once (e.g., whether we do single-site vs. two-site
+        DMRG).
```

Comment:
Order of parameters should match the order in the `__init__(...)` definition.
As mentioned in other places as well, I think it is redundant to include `EffectiveH` as a argument to __init__().